src/amo/utils/index.js

```diff
@@ -65,7 +65,10 @@ export const checkInternalURL = ({
 |}): { isInternal: boolean, relativeURL: string } => {
   const baseURL = _config.get('baseURL');
   const urlParts = url.parse(urlString, true);
-  const isInternal = url.parse(baseURL).host === urlParts.host;
+  const isInternal =
+    // The double slash is for protocol-free URLs.
+    (urlString.startsWith('/') && !urlString.startsWith('//')) ||
+    url.parse(baseURL).host === urlParts.host;
```

Comment:
I believe there should be a check/variable before to determine whether a URL is absolute or relative. Then, if the URL is relative, it is therefore internal. If the URL is absolute, then the `host` must be checked to determine whether the absolute URL is `internal`.

I think that'd be more explicit and future-proof.

Reply:
Good idea. I have updated the patch.